Remove most of the now-unused API::SerializedScriptValue

achristensen07 opened the pull request:
#### bc52dccd1b20c7685c044197e8c9b732b3743d67
<pre>
Remove most of the now-unused API::SerializedScriptValue
<a href="https://bugs.webkit.org/show_bug.cgi?id=296718">https://bugs.webkit.org/show_bug.cgi?id=296718</a>
<a href="https://rdar.apple.com/157148357">rdar://157148357</a>

Reviewed by Brent Fulgham.

The only piece of it that is still used is the creation of a JS context
to be used for deserializing to JS values, and that had 3 copies of the same code.
This removes the unused code and reduces it to 1 copy of the same code.

As is tradition, when removing a file we have to fix a bunch of issues related
to unified source building.  WKWebView.mm and WKWebViewTesting.mm are now in different
compilation units.

* Source/WebKit/Shared/API/APISerializedScriptValue.cpp:
(API::SharedJSContextWK::singleton):
(API::valueToWKObject): Deleted.
(API::SerializedScriptValue::deserializeWK): Deleted.
* Source/WebKit/Shared/API/APISerializedScriptValue.h:
(API::SerializedScriptValue::create): Deleted.
(API::SerializedScriptValue::createFromWireBytes): Deleted.
(API::SerializedScriptValue::deserialize): Deleted.
(API::SerializedScriptValue::dataReference const): Deleted.
(API::SerializedScriptValue::internalRepresentation): Deleted.
(API::SerializedScriptValue::SerializedScriptValue): Deleted.
* Source/WebKit/Shared/API/c/WKSharedAPICast.h:
* Source/WebKit/Shared/InspectorExtensionTypes.h:
* Source/WebKit/Shared/glib/JavaScriptEvaluationResultGLib.cpp:
(WebKit::JavaScriptEvaluationResult::toJSC):
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/Cocoa/APISerializedScriptValueCocoa.mm: Removed.
* Source/WebKit/UIProcess/API/glib/APISerializedScriptValueGLib.cpp: Removed.
* Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkit_web_view_get_javascript_global_context):
* Source/WebKit/UIProcess/UserContent/WebScriptMessageHandler.h:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/298063@main">https://commits.webkit.org/298063@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7156b500f33f60c1701a3888ae3f58e933bbb913

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/114085 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33837 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/24295 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/120253 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/64849 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/34465 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/42397 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86704 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/41698 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117033 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27441 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102466 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67088 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26624 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63952 "Built successfully") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96808 "Found 1 new API test failure: TestWebKitAPI.GPUProcess.ExitsUnderMemoryPressureWebAudioCase (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20711 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/123479 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/41114 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30636 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95543 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/41493 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98675 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95326 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40460 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18270 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/37262 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18290 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40991 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/46495 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40612 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43912 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/42365 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (approved):
It's beautiful! I love deleting code. r=me

achristensen07 committed:
Remove most of the now-unused API::SerializedScriptValue
https://bugs.webkit.org/show_bug.cgi?id=296718
rdar://157148357

Reviewed by Brent Fulgham.

The only piece of it that is still used is the creation of a JS context
to be used for deserializing to JS values, and that had 3 copies of the same code.
This removes the unused code and reduces it to 1 copy of the same code.

As is tradition, when removing a file we have to fix a bunch of issues related
to unified source building.  WKWebView.mm and WKWebViewTesting.mm are now in different
compilation units.

* Source/WebKit/Shared/API/APISerializedScriptValue.cpp:
(API::SharedJSContextWK::singleton):
(API::valueToWKObject): Deleted.
(API::SerializedScriptValue::deserializeWK): Deleted.
* Source/WebKit/Shared/API/APISerializedScriptValue.h:
(API::SerializedScriptValue::create): Deleted.
(API::SerializedScriptValue::createFromWireBytes): Deleted.
(API::SerializedScriptValue::deserialize): Deleted.
(API::SerializedScriptValue::dataReference const): Deleted.
(API::SerializedScriptValue::internalRepresentation): Deleted.
(API::SerializedScriptValue::SerializedScriptValue): Deleted.
* Source/WebKit/Shared/API/c/WKSharedAPICast.h:
* Source/WebKit/Shared/InspectorExtensionTypes.h:
* Source/WebKit/Shared/glib/JavaScriptEvaluationResultGLib.cpp:
(WebKit::JavaScriptEvaluationResult::toJSC):
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/Cocoa/APISerializedScriptValueCocoa.mm: Removed.
* Source/WebKit/UIProcess/API/glib/APISerializedScriptValueGLib.cpp: Removed.
* Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkit_web_view_get_javascript_global_context):
* Source/WebKit/UIProcess/UserContent/WebScriptMessageHandler.h:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/298063@main
